rules/dotnet-upgrade-for-complex-projects/rule.md

```diff
@@ -14,29 +14,22 @@ related:
 created: 2023-07-16T23:08:53.979Z
 guid: 9de5ca88-a6aa-4fe5-af47-d6d2169cde86
 ---
-
 There's not 1 single thing that makes a .NET project complicated to migrate to the latest .NET framework. Generally though it's a combination of the following:
 
-- Long lived code base
-- Lots of external dependencies
-- Lots of application components (web, desktop, services, etc)
-- etc
+* Long lived code base
+* Lots of external dependencies
+* Lots of application components (web, desktop, services, etc)
+* etc
 
-Your first action should always be to use the [.NET Upgrade Assistant](https://dotnet.microsoft.com/en-us/platform/upgrade-assistant). You can read more about the tool at [Do you know how to modernize your .NET applications?](https://www.ssw.com.au/rules/dotnet-upgrade-assistant/)
+Your first action should always be to use the [.NET Upgrade Assistant](https://dotnet.microsoft.com/en-us/platform/upgrade-assistant). You can read more about the tool at [Do you know how to modernize your .NET applications?](https://www.ssw.com.au/rules/dotnet-upgrade-assistant/) It is strongly advised to follow this rule before progressing further, particularly if you haven't had the opportunity to do so yet. 
 
 Below you will find some tips and tricks to help you with your more complicated migrations.
 
```

Comment:
The complex migration has many steps and may take a long time. 
However, there are many benefits, like incremental migrations, better risk management and easier track of progress.
The migration starts with an evolutionary migration and later transitions toward strangler fig to maximise the benefits of both apporaches. Evolutionary migration makes sure that the existing code is more compatible with .NET 8, while the strangler fig is used to fully migrate the code.
This way, we have a codebase that for a period of time work in the old .NET Framework but is transitioning toward .NET 8. Most changes will be beneficial to both platforms and you'll be able to deploy those changes to production. You can see more on the rule [Do you know the different ways to modernize your application?]( https://www.ssw.com.au/rules/modernize-your-app/)